### ISSUE TITLE
Where are fp- links documents stored?

### ISSUE THREAD

**Issue by obilodeau** [asciidoctor/docgist#30]
Pretty amazing tool. I have to say I'm impressed! Real-time collaboration is a big plus that should be highlighted more I think.

Anyway, those interactive documents (with links that start with fp-...) are hosted in firebase, correct?

Where is that data stored? If I don't pay for that storage, what's the catch? Is it an API key shared with all docgist users or a free distributed database service? It's not documented.

I would like to store sensitive documents in my docgist install. Can it be self-hosted?

Thanks,

**Comment by mojavelinux** [comment]
You're correct that the data for the collaborative editor is hosted in Firebase (via Firepad). It has the same terms as a free Firebase account. But that's not the first problem. The first problem is that I did not set up the Firebase instance nor do I have access to it. Complicating matters further, the person who set up the account is no longer with us and Firebase is requiring action from the owner in order to keep the instance past May 15th (which isn't going to happen). See #33.

Needless to say, we need a new plan of action, one where a custodian of the Asciidoctor project maintains the account.

Once we do have the new account, we need to think about how it works. Like I said, there was never really any formal discussion about how it would work and what the terms would be. So that discussion needs to happen.

Currently, the Firepad documents are secret in the sense that each document has a random ID. But otherwise, they are public. All the documents are stored in a single instance. Due to obscurity of ownership, no one actually has access to those documents except through the interface.

> I would like to store sensitive documents in my docgist install. Can it be self-hosted?

Yes, you can put docgist on any static webserver and, if you want to use the collaborative editor, reconfigure it to point to a Firebase instance of your choice. Here's the configuration: https://github.com/asciidoctor/docgist/blob/master/index.html#L23
